File: 2022/Day 1/Part 2/day1_2.hs
import Data.List.Split (splitOn)
import Data.List (sort)

readInput :: String -> [Int]
readInput = map (sum . map read . lines) . splitOn "\n\n"

solve :: [Int] -> Int
solve = sum . take 3 . reverse . sort

main :: IO ()
main = interact (show . solve . readInput)

File: 2022/Day 1/Part 2/day1_2.py
elves = [0, 0, 0]

for calories in open(0).read().split("\n\n"):
    curr = sum(map(int, calories.split()))
    for i in range(3):
        if curr > elves[i]:
            elves[i], curr = curr, elves[i]

print(sum(elves))